inspirehep/modules/workflows/views.py

```diff
@@ -376,3 +403,140 @@ def robotupload_callback():
         responses[recid] = response
 
     return jsonify(responses)
+
+
+def _validate_workflow_schema(workflow_data):
+    """Validate the ``metadata`` against the ``hep`` JSONSchema.
+
+    Args:
+        workflow_data (dist): the workflow dict.
+
+    Raises:
+        CallbackValidationError: if the workflow ``metadata`` is not valid
+            against ``hep`` JSONSchema.
+    """
+
+    # Check for validation errors
+    try:
+        validate(schema='hep', data=workflow_data['metadata'])
```

Comment:
It can stay for now, but this machinery will also be useful to fix author submissions/updates that failed validation, and the fact that we're hardcoding the schema prevents this...